src/extensions/replay/triggerMatching.ts

```diff
@@ -0,0 +1,314 @@
+            this.instance.onFeatureFlags((_flags, variants) => {
+                const flagIsPresent = isObject(variants) && linkedFlag in variants
+                const linkedFlagMatches = linkedVariant ? variants[linkedFlag] === linkedVariant : flagIsPresent
+                if (linkedFlagMatches) {
+                    onStarted(linkedFlag, linkedVariant)
+                }
+                this.linkedFlagSeen = linkedFlagMatches
+            })
+        }
+    }
+}
+
+export class EventTriggerMatching implements TriggerStatusMatching {
+    _eventTriggers: string[] = []
+
+    constructor(private readonly instance: PostHog) {}
+
+    onRemoteConfig(response: RemoteConfig) {
+        this._eventTriggers = response.sessionRecording?.eventTriggers || []
+    }
+
+    private eventTriggerStatus(sessionId: string): TriggerStatus {
+        if (this._eventTriggers.length === 0) {
+            return 'trigger_disabled'
+        }
+
+        const currentTriggerSession = this.instance?.get_property(SESSION_RECORDING_EVENT_TRIGGER_ACTIVATED_SESSION)
+        return currentTriggerSession === sessionId ? 'trigger_activated' : 'trigger_pending'
+    }
+
+    triggerStatus(sessionId: string): TriggerStatus {
+        const eventTriggerStatus = this.eventTriggerStatus(sessionId)
+        return eventTriggerStatus === 'trigger_activated'
+            ? 'trigger_activated'
+            : eventTriggerStatus === 'trigger_pending'
+              ? 'trigger_pending'
+              : 'trigger_disabled'
+    }
+}
+
+export interface RecordingTriggersStatus {
+    get receivedDecide(): boolean
+    get isRecordingEnabled(): false | true | undefined
+    get isSampled(): false | true | null
+    get urlTriggerMatching(): URLTriggerMatching
+    get eventTriggerMatching(): EventTriggerMatching
+    get linkedFlagMatching(): LinkedFlagMatching
+    get sessionId(): string
+}
+
+// we need a no-op matcher before we can lazy load the other matches, since all matchers wait on remote config anyway
+export function nullMatchSessionRecordingStatus(triggersStatus: RecordingTriggersStatus): SessionRecordingStatus {
+    if (!triggersStatus.isRecordingEnabled) {
+        return 'disabled'
+    }
+
+    return 'buffering'
+}
+
+export function anyMatchSessionRecordingStatus(triggersStatus: RecordingTriggersStatus): SessionRecordingStatus {
+    if (!triggersStatus.receivedDecide) {
+        return 'buffering'
+    }
+
+    if (!triggersStatus.isRecordingEnabled) {
+        return 'disabled'
+    }
+
+    if (triggersStatus.urlTriggerMatching.urlBlocked) {
+        return 'paused'
+    }
+
+    const sampledActive = triggersStatus.isSampled === true
+    const triggerMatches = new OrTriggerMatching([
+        triggersStatus.eventTriggerMatching,
+        triggersStatus.urlTriggerMatching,
+        triggersStatus.linkedFlagMatching,
+    ]).triggerStatus(triggersStatus.sessionId)
+
+    if (sampledActive || triggerMatches === 'trigger_activated') {
```

Comment:
I'm confused about this check. 

- I suggest `sampled` used here in a positive way (like, got in the sample, record)? As lower there's a `triggersStatus.isSampled === false` => `disabled` check.
- As `sampledActive` is checked for truthiness - what's the point of another check `sampledActive ? 'sampled' : 'active'` if we know it always true?

Reply:
ah, historically we report the status of the recorder by saying "sampled" when it was sampled instead of active

so inside the block we could be in because of sampling or because of triggers so we have to check again so we report the right status

iirc it's only for debug and i think i'd like to make deciding and reporting on deciding more explicit

but, here you could make it two ifs

if (sampledActive) return 'sampled'
if (triggerMatches === 'trigger_activated') return 'active

for the same result i guess